gradoop-flink/src/main/java/org/gradoop/flink/io/impl/text/GDLEncoder.java

```diff
@@ -0,0 +1,254 @@
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Encodes data structures into GDL format.
+ */
+public class GDLEncoder {
+  /**
+   * Marks the beginning of the definition of vertices and edges.
+   */
+  private static final String GRAPH_ELEMENTS_DEFINITION_START = "[";
+  /**
+   * Marks the end of the definition of vertices and edges.
+   */
+  private static final String GRAPH_ELEMENTS_DEFINITION_END = "]";
+  /**
+   * Seperates the id and label of graph, vertex and edge.
+   */
+  private static final String ID_LABEL_SEPERATOR = ":";
+  /**
+   * Graph variable prefix, because Gradoop can't parse GDL variable starting with a number.
+   */
+  private static final String GRAPH_VARIABLE_PREFIX = "g";
+  /**
+   * vertex variable prefix, because Gradoop can't parse GDL variable starting with a number.
+   */
+  private static final String VERTEX_VARIABLE_PREFIX = "v";
+  /**
+   * Seperates properties.
+   */
+  private static final String PROPERTIES_PREFIX = "{";
+  /**
+   * Seperates properties.
+   */
+  private static final String PROPERTIES_SUFFIX = "}";
+  /**
+   * Seperates properties.
+   */
+  private static final String PROPERTIES_SEPERATOR = ",";
+  /**
+   * Seperates key and value for properties.
+   */
+  private static final String KEY_VALUE_SEPERATOR = ":";
+  /**
+   * Suffix for GDL double representation.
+   */
+  private static final String DOUBLE_SUFFIX = "d";
+  /**
+   * Suffix for GDL float representation.
+   */
+  private static final String FLOAT_SUFFIX = "f";
+  /**
+   * Suffix for GDL long representation.
+   */
+  private static final String LONG_SUFFIX = "L";
+  /**
+   * GDL null representation.
+   */
+  private static final String NULL_STRING = "NULL";
+  /**
+   * GDL string prefix
+   */
+  private static final String STRING_PREFIX = "\"";
+  /**
+   * GDL string suffix
+   */
+  private static final String STRING_SUFFIX = "\"";
+
+  /**
+   * Graph head to encode.
+   */
+  private GraphHead graphHead;
+  /**
+   * Vertices to encode.
+   */
+  private List<Vertex> vertices;
+  /**
+   * Edges to encode.
+   */
+  private List<Edge> edges;
+
+  /**
+   *
+   * @param graphHead graph head that should be encoded
+   * @param vertices vertices that should be encoded
+   * @param edges edges that should be encoded
+   */
+  public GDLEncoder(GraphHead graphHead, List<Vertex> vertices, List<Edge> edges) {
+    this.graphHead = graphHead;
+    this.vertices = vertices;
+    this.edges = edges;
+  }
+
+  /**
+   * Transforms the graph that is represented by graph head, vertices and edges into a GDL string.
+   * @return GDL formatted String.
+   */
+  public String graphToGDLString() {
+
+    Map<GradoopId, String> idToVertexName = getVertexNameMapping(vertices);
+
+
+    String graphHeadString = graphHeadToGDLString(graphHead);
+
+    String verticesString = vertices.stream()
+      .map(v -> vertexToGDLString(v, idToVertexName.get(v.getId())))
+      .collect(Collectors.joining("\n"));
```

Comment:
"\n" is unix specific -> System.getProperty("line.separator") gets you the system line separator.